test/androidTest/java/org/thoughtcrime/securesms/PinnedMessageEspressoTest.java

```diff
@@ -4,34 +4,67 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import android.content.Intent;
 import android.support.test.runner.AndroidJUnit4;
 import android.support.test.rule.ActivityTestRule;
 import android.support.test.filters.LargeTest;
+
 import static android.support.test.espresso.Espresso.onView;
-import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
-import static android.support.test.espresso.assertion.ViewAssertions.matches;
 
 @RunWith(AndroidJUnit4.class)
 @LargeTest
 public class PinnedMessageEspressoTest {
 
     @Rule
-    public ActivityTestRule<PinnedMessageActivity> pinnedActivityRule =
-            new ActivityTestRule(PinnedMessageActivity.class, true, false);
+    public ActivityTestRule<ConversationListActivity> mainActivityRule =
+            new ActivityTestRule(ConversationListActivity.class, true, false);
 
     @Test
     public void pageExists() {
-        Intent intent = new Intent();
+        EspressoHelper helper = new EspressoHelper(mainActivityRule);
+
+        helper
+            .createConversation()
+            .goPinned()
+            .done();
+    }
+
+    @Test
+    public void canPinMessages() {
+        EspressoHelper helper = new EspressoHelper(mainActivityRule);
 
-        intent.putExtra("address", "514 123 4567");
-        intent.putExtra("THREADID", 1);
+        String testString = helper.randString();
+
+        helper
+            .createConversation()
+                .sendMessage(testString)
+                .selectMessage(1)
+                .pinSelectedMessage()
+            .goPinned()
+                .findText(testString)
+            .done();
+    }
 
-        pinnedActivityRule.launchActivity(intent);
 
-        // onView(withId(R.id.pinned_message_body)).check(matches(isDisplayed()));
+    @Test
+    public void canUnPinMessages() {
+        EspressoHelper helper = new EspressoHelper(mainActivityRule);
 
-        onView(withText("Signal needs access to your contacts and media in order to connect with friends, exchange messages, and make secure calls")).check(matches(isDisplayed()));
+        String testString = helper.randString();
+
+        helper
+            .createConversation()
+                .sendMessage(testString)
+                .selectMessage()
+                .pinSelectedMessage()
+            .goPinned()
+                .findText(testString)
+            .goConversation()
+                .selectMessage()
+                .unpinSelectedMessage()
+            .goPinned()
+                .not_findText(testString)
+            .done();
     }
+
 }
```

Comment:
I still don't understand how you bypassed the registration

Reply:
only part that needs to be done manually